cdap-watchdog/src/test/java/io/cdap/cdap/logging/clean/LogCleanerTest.java

```diff
@@ -227,33 +231,41 @@ public void testLogFolderCleanup(int folderCleanupBatchSize,
   }
 
   @Test
-  public void testLogFolderCleanupExclusion() throws IOException {
+  public void testLogFolderCleanupExclusion() throws IOException, InterruptedException {
     // Create log folders and files
     TransactionRunner transactionRunner = injector.getInstance(TransactionRunner.class);
     FileMetadataCleaner fileMetadataCleaner = new FileMetadataCleaner(transactionRunner);
     LocationFactory locationFactory = new LocalLocationFactory(TMP_FOLDER.newFolder());
-    Location logFolder = locationFactory.create("logs");
+    Location logFolder = locationFactory.create("logs" + System.currentTimeMillis());
     logFolder.mkdirs();
 
-    for (int day = 0; day < 4; day++) {
-      long time = System.currentTimeMillis();
-      long millisInDay = 1000 * 60 * 60 * 24;
-      String date = new SimpleDateFormat("yyyy-MM-dd").format(new Date(time - day * millisInDay));
-      Location location = logFolder.append(date);
-      location.mkdirs();
+    for (int dayOffset = 5; dayOffset >= 3; dayOffset--) {
+      createFolderWithDateFormat(logFolder, dayOffset);
+    }
+
+    Thread.sleep(5000);
+    for (int dayOffset = 2; dayOffset >= 0; dayOffset--) {
+      createFolderWithDateFormat(logFolder, dayOffset);
     }
 
     // Run log folder cleaner
     LogCleaner logCleaner = new LogCleaner(fileMetadataCleaner,
         locationFactory,
         logFolder,
-        100,
+        2000,
         10,
         10);
     logCleaner.run();
 
     // Assert for the remaining folder count
-    Assert.assertEquals(2, logFolder.list().size());
+    Assert.assertEquals(3, logFolder.list().size());
+  }
+
+  private void createFolderWithDateFormat(Location logFolder, int dayOffset) throws IOException {
+    long time = System.currentTimeMillis();
+    String date = new SimpleDateFormat("yyyy-MM-dd").format(new Date(time - dayOffset * MILLIS_IN_DAY));
```

Comment:
Should use the `LogFileManager.formatLogDirectoryName`?

Reply:
Done.